prestodb/presto
# Add Session Properties support

## Description
Add a new widget in the `SQL Client` page to list all session properties. Users can edit the session properties if needed and then submit the SQL queries. In this change, presto-js-client is upgraded to 1.0.0 which improves the error handling.

related to #21062 item 8

## Motivation and Context
After adding the `SQL Client` page, one of the suggestions is to [add session properties support](https://github.com/prestodb/presto/issues/21062#issuecomment-1879043677). This feature would make the SQL Client more useful.

## Impact
When running the SQL queries from the `SQL Client` page, users can update session properties. 

## Test Plan
Run the new UI with the following scenarios:
- normal SQL queries to make sure the presto-js-client upgrade
- check the `session properties` page and verify the session properties table
- update one or two session properties and run a query to see if the session properties are carried with the SQL query.

## Contributor checklist


## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```
